fixed to Validate ID was not modified

moishce opened the pull request:
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/29413

## Description
fixed to Validate ID was not modified

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

ShahafBenYakir commented:
Please add explanation on how this fixes the issue.
Also, add a test for this scenario.

moishce commented:
After I fixed it to look for the old file then it failed on the build on validating:
_id_has_no_beta_substring, _name_has_no_beta_substring
and he checked it only for modified files
So I fixed it that would do it only for new files

ShahafBenYakir commented on demisto_sdk/commands/common/hook_validations/integration.py:
```diff
@@ -351,15 +351,15 @@ def is_changed_subtype(self):
     def is_valid_beta(self):
         # type: () -> bool
         """Validate that that beta integration has correct beta attributes"""
```
```suggestion
        """Validate that beta integration has correct beta attributes"""
```

ShahafBenYakir commented on demisto_sdk/commands/common/hook_validations/integration.py:
```diff
@@ -351,15 +351,15 @@ def is_changed_subtype(self):
         if not all([self._is_display_contains_beta(), self._has_beta_param()]):
             self.is_valid = False
-            return False
-        if self.old_file:
+            valid_status = False
+        if not self.old_file:
             if not all([self._id_has_no_beta_substring(), self._name_has_no_beta_substring()]):
                 self.is_valid = False
-                return False
-        return True
+                valid_status = False
+        return valid_status
```
I think that after adding the `not self.old_file` this should work, so why do we need the `valid_status` argument?

moishce replied:
I want to return true or false only at the end of the function so that it passes all the tests and does not return in the first failure

ShahafBenYakir replied:
Mmmmm... why? if the first check fails the status will be `False` no matter what will happen in the second check. Is there something else happening in the second check that I am missing?

moishce replied:
Yes, inside the functions it puts the error into the handle_error and it prints the error, and if it finishes in the first function then only the error of the first will return

ShahafBenYakir replied:
Cool I see

ShahafBenYakir commented on demisto_sdk/tests/test_files/valid-beta-integration.yml:
```diff
@@ -1,6 +1,6 @@
 category: Messaging
 commonfields:
-  id: Telegram_Beta
+  id: Telegram
```
This is for testing right?

moishce replied:
right

ShahafBenYakir replied:
If we are done testing, let's revert this.

roysagi reviewed (commented):
Please add a changelog entry for this.

ShahafBenYakir commented on demisto_sdk/commands/common/hook_validations/structure.py:
```diff
@@ -61,7 +62,7 @@ def __init__(self, file_path, is_new_file=False, old_file_path=None, predefined_
         self.current_file = self.load_data_from_file()
         self.fromversion = fromversion
-        if is_new_file or predefined_scheme:
+        if is_new_file or not is_file_path_in_pack(self.file_path):
```
Add a comment on what this condiition checks

ShahafBenYakir commented on demisto_sdk/commands/validate/tests/validators_test.py:
```diff
@@ -381,6 +381,40 @@ def test_run_all_validations_on_file(self, _, file_path):
+    @pytest.mark.parametrize('file_path', INVALID_FILES_PATHS_FOR_ALL_VALIDATIONS)
+    @patch.object(ImageValidator, 'is_valid', return_value=True)
+    def test_run_all_validations_on_file_failed(self, _, file_path):
+        """
+        Given
+        - A invalid file in packs or beta integration
+
+        When
+        - running run_all_validations_on_file on that file
+
+        Then
+        -  The file will be validated and failed
+        """
```
```suggestion
        """
        Given
        - An invalid file inside packs or a beta integration

        When
        - running run_all_validations_on_file on that file

        Then
        -  The file will be validated and failed
        """
```
I don't see we tested a beta integration file, we should... no?

ShahafBenYakir reviewed (commented):
Let's add a test for chagning an item ID

ShahafBenYakir commented on demisto_sdk/commands/validate/tests/validators_test.py:
```diff
@@ -381,6 +381,64 @@ def test_run_all_validations_on_file(self, _, file_path):
+    def test_run_all_validations_on_file_failed(self, _, file_path):
+        """
+        Given
+        - An invalid file inside packs or an invalid beta integration
```
```suggestion
        - An invalid file inside a pack
```

ShahafBenYakir commented on demisto_sdk/commands/validate/tests/validators_test.py:
```diff
@@ -381,6 +381,64 @@ def test_run_all_validations_on_file(self, _, file_path):
+    def test_run_all_validations_on_file_white_modified_id(self, mocker):
+        """
+        Given
+        - file in packs thet id was modified
+
+        When
+        - running run_all_validations_on_file on that file
+
+        Then
+        -  The file will be validated and failed Because the id is not equal to the old file
+        """
```
```suggestion
        - Integration with a modified ID.

        When
        - running 'run_all_validations_on_file' on that file.

        Then
        -  The file will fail validation and because its id changed.
        """
```
